packages/teams-js/test/public/dialog.spec.ts

```diff
@@ -524,138 +524,177 @@ describe('Dialog', () => {
+                expect(message).not.toBeUndefined();
+                expect(message.args).toStrictEqual(['exampleMessage']);
+              });
+            } else {
+              it(`FRAMELESS: should not allow calls from ${frameContext} context`, async () => {
+                await utils.initializeWithContext(frameContext);
+                expect(() => dialog.url.parentCommunication.sendMessageToParentFromDialog('message')).toThrowError(
+                  `This call is only allowed in following contexts: ${JSON.stringify(
+                    allowedContexts,
+                  )}. Current context: "${frameContext}".`,
+                );
+              });
+            }
+          });
+      });
+      describe('registerOnMessageFromParent', () => {
+        const allowedContexts = [FrameContexts.task];
+
+        it('should not allow calls before initialization', () => {
+          expect.assertions(1);
+          expect(() => dialog.url.parentCommunication.registerOnMessageFromParent(emptyCallback)).toThrowError(
+            errorLibraryNotInitialized,
+          );
+        });
+
+        Object.keys(FrameContexts)
+          .map((k) => FrameContexts[k])
+          .forEach((frameContext) => {
+            if (frameContext === FrameContexts.task) {
+              it(`FRAMELESS: should throw error when dialog is not supported in ${frameContext} context`, async () => {
+                await utils.initializeWithContext(frameContext);
+                utils.setRuntimeConfig({ apiVersion: 4, supports: {} });
+                expect.assertions(1);
+                try {
+                  dialog.url.parentCommunication.registerOnMessageFromParent(emptyCallback);
+                } catch (e) {
+                  expect(e).toEqual(errorNotSupportedOnPlatform);
+                }
+              });
+
+              it(`FRAMELESS: should initiate the registration call: ${frameContext}`, async () => {
+                await utils.initializeWithContext(frameContext);
+                dialog.url.parentCommunication.registerOnMessageFromParent(emptyCallback);
+                const message = utils.findMessageByFunc('registerHandler');
+                expect(message).not.toBeUndefined();
+                expect(message.args).toStrictEqual(['messageForChild']);
+              });
+            } else {
+              it(`FRAMELESS: should not allow calls from ${frameContext} context`, async () => {
+                await utils.initializeWithContext(frameContext);
+                expect(() => dialog.url.parentCommunication.registerOnMessageFromParent(emptyCallback)).toThrowError(
+                  `This call is only allowed in following contexts: ${JSON.stringify(
+                    allowedContexts,
+                  )}. Current context: "${frameContext}".`,
+                );
+              });
+            }
+          });
+      });
+      describe('dialog.url.parentCommunication.isSupported function', () => {
```

Comment:
Should there be a test which validates that if apiVersion is v3 and dialog.url is supported, then isSupported returns true?

What about a test that validates that if apiVersion is v3 and dialog.url is not supported then isSupported returns false?

Reply:
Added these two tests.